a3p-integration/proposals/b:wallet-factory2/performActions.ts

```diff
@@ -0,0 +1,340 @@
+#!/usr/bin/env tsx
+
+import { execFileSync } from 'child_process'; // TODO: use execa
+import * as fsp from 'fs/promises';
+import * as path from 'path';
+import * as process from 'process';
+
+import { ZipReader } from '@endo/zip';
+import dbOpen from 'better-sqlite3';
+
+import assert from 'assert';
+import { makeAgd } from '@agoric/synthetic-chain/src/lib/agd-lib.js';
+import { agoric } from '@agoric/synthetic-chain/src/lib/cliHelper.js';
+import {
+  bundleDetail,
+  ensureISTForInstall,
+  flags,
+  getContractInfo,
+  getUser,
+  loadedBundleIds,
+  type ProposalInfo,
+  txAbbr,
+  voteLatestProposalAndWait,
+} from '@agoric/synthetic-chain/src/lib/commonUpgradeHelpers.js';
+import { dbTool } from '@agoric/synthetic-chain/src/lib/vat-status.js';
+import { type WebCache } from '@agoric/synthetic-chain/src/lib/webAsset.js';
+
+// TODO move into core-eval-support
+const readSubmissions = async () => {
+  const files = await fsp.readdir('submission');
+  const names = files.filter(f => f.endsWith('.js')).map(f => f.slice(0, -3));
+  const buildAssets = {} as Record<string, ProposalInfo>;
+  for (const name of names) {
+    const evals = [{ permit: `${name}-permit.json`, script: `${name}.js` }];
+    // eslint-disable-next-line no-await-in-loop
+    const content = await fsp.readFile(`submission/${name}.js`, 'utf8');
+    const bundleIds = content.matchAll(/b1-[a-z0-9]+/g);
+    const bundles = Array.from(bundleIds).map(id => `${id}.json`);
+    buildAssets[name] = { evals, bundles };
+  }
+  return buildAssets;
+};
+
+/**
+ * URLs of assets, including bundle hashes (to be) agreed by BLD stakers
+ */
+const assetInfo = {
+  repo: {
+    name: 'wallet-factory1',
+    description: 'first upgrade of wallet factory vat',
+  },
+  buildAssets: await readSubmissions(),
+};
+
+const staticConfig = {
+  deposit: '10000000ubld', // 10 BLD
+  installer: 'gov1', // as in: agd keys show gov1
+  proposer: 'validator',
+  collateralPrice: 6, // conservatively low price. TODO: look up
+  swingstorePath: '~/.agoric/data/agoric/swingstore.sqlite',
+  title: assetInfo.repo.name,
+  description: assetInfo.repo.description,
+  buildInfo: Object.values(assetInfo.buildAssets),
+};
+
+const makeFakeWebCache = (base: string): WebCache => {
+  return {
+    getText(segment: string) {
+      return fsp.readFile(path.join(base, segment), 'utf8');
+    },
+    async storedPath(segment: string) {
+      return path.join(base, segment);
+    },
+    async size(segment: string) {
+      const info = await fsp.stat(path.join(base, segment));
+      return info.size;
+    },
+    toString() {
+      return 'fake web cache';
+    },
+    async remove() {
+      console.warn('noop remove');
+    },
+  };
+};
+
+/** Provide access to the outside world via context. */
+const makeTestContext = async () => {
+  // assume filenames don't overlap
+  const bundleAssets = makeFakeWebCache('submission');
```

Comment:
if you're going to keep this part of `performActions.ts` then consider renaming the submission directory and this string that's looking for it. 

But I think you could simplify this file to,
```js
import { evalBundles } from '@agoric/synthetic-chain/src/lib/core-eval.js`;
evalBundles('zcfProber-submission');
```

Oh, I guess there's already support for specifying a directory path. 🤦

Reply:
I reran `yarn; yarn build`, but I still don't see `'@agoric/synthetic-chain/src/lib/core-eval.js`;`. I looked in a3p as well, and it doesn't appear to be in `packages/synthetic-chain/src/lib`, and I've just pulled again.